Mail2Bug/WorkItemManagement/TFSWorkItemManager.cs

```diff
@@ -304,6 +304,35 @@ public void ModifyWorkItem(int workItemId, string comment, Dictionary<string, st
             workItem.Save();
         }
 
+        public SimpleWorkItem GetWorkItem(int workItemId)
+        {
+            if (workItemId <= 0) return null;
+            var tfsWorkItem = _tfsStore.GetWorkItem(workItemId);
+
+            var workItem = new SimpleWorkItem()
+            {
+                Id = tfsWorkItem.Id,
+                Title = tfsWorkItem.Title,
+                AssignedTo = tfsWorkItem.Fields["Assigned To"].Value as string,
+                State = tfsWorkItem.State,
+                Fields = AsDictionary(tfsWorkItem.Fields)
+            };
+
+            return workItem;
+        }
+
+        private Dictionary<string, string> AsDictionary(FieldCollection fields)
```

Comment:
Copying all the fields to a new dictionary is risky performance-wise. If the item has a lot of fields, you will be copying all of them, even though we'll probably use at most 1 or 2 of them (and more often than not, none of them).

Alternative solution
1. Change the interface to provide a 'GetFieldValue(string fieldName)' method rather than exposing the dictionary
2. Hold a (private) reference to the original TFS item instead of the dictionary
3. Implement GetFieldValue by retrieving the field with that name from the item
That way, you'll avoid the copying of anything that's not used
We can make ID and Title available that way too (either by querying them by name from the item, if that works, or by handling them explicitly in the code if it doesn't work).